doc.go

```diff
@@ -50,7 +50,18 @@ func (d *Document) fieldIndex(name string) (int, error) {
 func (d *Document) newField(field index.Field) {
 	d.fieldIndexes[field.Name()] = len(d.fieldTokenFreqs)
 	d.fieldNames = append(d.fieldNames, field.Name())
-	d.fieldTokenFreqs = append(d.fieldTokenFreqs, field.AnalyzedTokenFrequencies())
+	af := field.AnalyzedTokenFrequencies()
+
+	// bleve analysis will leave field empty for non-composite fields, fix that here
```

Comment:
Hmm, I see that "field" isn't set for TokenLocation within the TokenFrequency() api which calculates this. I forget - what is the reason we don't set the field name there?

Reply:
The best answer I can give you is that we've just always done it that way.

Upsidedown compensates for it here: https://github.com/blevesearch/bleve/blob/36861ddf9f6d58da663671ac86da182b1a45d786/index/upsidedown/upsidedown.go#L745-L751

And zap/zapx does it here: https://github.com/blevesearch/zapx/blob/a6504989eeb62fda244e15dd259c4b2484eb672d/new.go#L465-L467

I have some vague recollection, that at some time way in the past, either we didn't have composite fields yet, or the term vector information didn't separately store the field (for non-composite fields it is redundant).  Perhaps our thinking was that by passing the information in this way, a smarter encoding might be able to take advantage of it (ie, save space don't record field for cases where you already know it).  If we modified the analysis to always set the field, we wouldn't be able to make such a determination.